hastory-server/src/Data/Hastory/Server/Handler/Users.hs

```diff
@@ -0,0 +1,12 @@
+{-# LANGUAGE RecordWildCards #-}
+
+module Data.Hastory.Server.Handler.Users where
+
+import Data.Hastory.Server.Handler.Import
+
+createUserHandler :: ServerSettings -> UserForm -> Handler UserId
+createUserHandler ServerSettings {..} UserForm {..} = do
+  userNameCount <- runDB (count [UserName ==. userFormUserName]) _ssDbPool
+  when (userNameCount > 0) (throwError err400)
```

Comment:
It's probably best to use evidence instead of a boolean.
So use `getBy $ UniqueUsername userFormUserName` and then case-match on the `Maybe (Entity User)`.

Reply:
I also don't know how case-insensitivity will interact with uniqueness constraints.
If you really want the case-insensitivity constraint, then I suggest writing a test that tries to make a user with usernam `foo` and then one with username `FOO` or `Foo`

Reply:
Great point on using the evidence vs. the boolean. I did just. I wrote out property specs in `UsersSpec` module. Let me know what you think. I had to update the persistent entity definition for user by adding `sqltype=`:

```
User
    name Username sqltype="VARCHAR COLLATE NOCASE"
    hashedPassword (PasswordHash Bcrypt)
    UniqueUsername name
```

Apparently that is how you do case insensitivity with sqlite; I'm more familiar with postgres, but the tests give me confidence in the behavior.